src/Features/Core/Portable/IncrementalCaches/SymbolTreeInfoIncrementalAnalyzerProvider.cs

```diff
@@ -198,13 +198,30 @@ private async Task UpdateSymbolTreeInfoAsync(Project project, CancellationToken
                 Debug.Assert(SupportAnalysis(project));
 
                 // Produce the indices for the source and metadata symbols in parallel.
-                var tasks = new List<Task>
+                var isRemoteWorkspace = project.Solution.Workspace.Kind == WorkspaceKind.RemoteWorkspace;
+                Task[] tasks = isRemoteWorkspace
+                    ? UpdateSymbolTreeInfoInServiceProcessAsync(project, cancellationToken)
+                    : UpdateSymbolTreeInfoInCurrentProcessAsync(project, cancellationToken);
```

Comment:
i'm a little confused by what this approach is trying to do.  It seems like the only difference between the two is if you do an Additional Task.Run.  But it's unclear why that makes a substantive difference.  And, if it doesn't, it's unclear why we woudln't want that to be what we *always* do.  Can you provide some clarity here?

Reply:
Oh, i see.  The goal is to be able to do more in parallel in the remote process, while trying not to overload teh VS process if we're running in proc.

Would def be good to comment this stuff.

Reply:
Will add one.

Reply:
Added comments in the methods.